guide/interactions/select-menus.md

```diff
@@ -0,0 +1,341 @@
+:::
+
+Restart your bot and then send the command to a channel your bot has access to. If all goes well, you should see something like this:
+
+<!--- vue-discord-message doesn't yet have support for inline replies/interactions/ephemeral messages/components -->
+<div is="discord-messages">
+	<discord-message profile="user">
+		/ping
+	</discord-message>
+	<discord-message profile="bot">
+		Pong! (+ components)
+	</discord-message>
+</div>
+
+You can also send message components within an ephemeral response or alongside message embeds.
+
+```js {1,26-30,32}
+const { MessageActionRow, MessageSelectMenu, MessageEmbed } = require('discord.js');
+
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		const row = new MessageActionRow()
+			.addComponents(
+				new MessageSelectMenu()
+					.setCustomId('select')
+					.setPlaceholder('Nothing selected')
+					.addOptions([
+						{
+							label: 'Select me',
+							description: 'This is a description',
+							value: 'first_option',
+						},
+						{
+							label: 'You can select me too',
+							description: 'This is also a description',
+							value: 'second_option',
+						},
+					]),
+			);
+
+		const embed = new MessageEmbed()
+			.setColor('#0099ff')
+			.setTitle('Some title')
+			.setURL('https://discord.js.org/')
+			.setDescription('Some description here');
+
+		await interaction.reply({ content: 'Pong!', ephemeral: true, embeds: [embed], components: [row] });
+	}
+});
+```
+
+Restart your bot and then send the command to a channel your bot has access to. If all goes well, you should see something like this:
+
+<!--- vue-discord-message doesn't yet have support for inline replies/interactions/ephemeral messages/components -->
+<div is="discord-messages">
+	<discord-message profile="bot">
+		Pong! (ephemeral + components)
+		<discord-embed
+			slot="embeds"
+			color="#0099ff"
+			title="Some title"
+			url="https://discord.js.org/"
+		>
+			Some description here
+		</discord-embed>
+	</discord-message>
+</div>
+
+Additionally, if you don't want to construct an `ActionRow` every time, you can also pass an array of arrays containing components like this:
+
+```js {7-21,23}
+const { MessageActionRow, MessageSelectMenu, MessageEmbed } = require('discord.js');
+
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		const menu = new MessageSelectMenu()
+			.setCustomId('select')
+			.setPlaceholder('Nothing selected')
+			.addOptions([
+				{
+					label: 'Select me',
+					description: 'This is a description',
+					value: 'first_selection',
+				},
+				{
+					label: 'You can select me too',
+					description: 'This is also a description',
+					value: 'second_selection',
+				},
+			]);
+
+		await interaction.reply({ content: 'Pong!', components: [[menu]] });
+	}
+});
```

Comment:
```suggestion
<!-- eslint-skip -->

```js
const menu = new MessageSelectMenu()
	.setCustomId('select')
	.setPlaceholder('Nothing selected')
	.addOptions([
		{
			label: 'Select me',
			description: 'This is a description',
			value: 'first_option',
		},
		{
			label: 'You can select me too',
			description: 'This is also a description',
			value: 'second_option',
		},
	]);

await interaction.reply({ content: 'Pong!', components: [[menu]] });
```